### ISSUE TITLE
Update `trails-viz` to only pull data from relevant project folder in `trails-viz-data`

### ISSUE THREAD

**Issue by sgwinder** [OutdoorRD/trails-viz#187]
We have a couple of cases now where a single `siteid` can be found in multiple project polygon files. For example, many of the `OKW_000` polygons are included in both the `east-cascades` polygon file ([`OKW_NVUM_polygons.geojson`](https://github.com/OutdoorRD/east-cascades/blob/main/data/gis/OKW_NVUM_Polygons.geojson)) and in the `coronado` polygon file ([`OKW_MBS_Coronado_Trails.geojson`](https://github.com/OutdoorRD/nvum-coronado/blob/master/data/gis/OKW_MBS_Coronado_Trails.geojson)).

This appears to be causing a problem in `trails-viz`, now that we have launched dashboards for both the Coronado and the East Cascades projects. Specifically, it looks like the data from `trails-viz-data/projects/Coronado/viz_model_mmm.csv` and `viz_model_www.csv` in the same folder are being read for the [`east-cascades` dashboard](https://trailtrends.outdoorrd.org/dashboard/EastCascades). These data should not be pulled into this dashboard, even though their siteids match. Instead, better behavior would be for the `east-cascades` dashboard to ONLY read data from the `trails-viz-data/projects/EastCascades` folder. 

Currently, I think the EastCascades dashboard is reading from both folders. There are a couple of hints, but the biggest one is that the data in `trails-viz-data/projects/EastCascades` does not include 2022, so the EastCascades dashboard also should not include 2022. It's leading to strange plots like this one: 
<img width="931" alt="image" src="https://github.com/OutdoorRD/trails-viz/assets/46452196/09ae7600-318d-45df-9620-46b44b091a84">

@EmiliaH and I are presenting to the OKW leadership team on Tuesday, and it would really be nice if the EastCascades dashboard were functioning again before that point. I think a patch would be to delete all of the `OKW_000` rows from the data in `trails-viz-data/projects/Coronado`, but if a more durable solution isn't too hard then that would be fantastic. @jblehr , what do you think? Do you have time to look into this in the next couple of days?

**Comment by jblehr** [comment]
@sgwinder, yes, I can take a look at this before the end of this week. Thanks for the detailed write-up of the issue.

**Comment by jblehr** [comment]
@sgwinder, I have some updates here:

First, I confirmed your theory: When I remove the `OKW` rows from the `trails-viz-data/Coronado/...csv` files, the dashboard operates correctly.

Next, the project/site distinction is pretty core throughout the dashboard code, in both the frontend and the backend. I tried a few different things, but am not sure the best approach to solving this. It will also depend on how we plan to use the project/site distinction moving forwards. I think it will take a bit more thought and testing before implementing something worthwhile.

Based on this, to fix this before Tuesday, I'd suggest making the patch for now (removing the OKW/MBS rows from the `Coronado` folder).  I can do this directly on the VM, but I'd suggest adding some code or documenting this somewhere in the `viz-model` repo that produces these files, too.  How does that sound?

**Comment by jblehr** [comment]
Adding my notes here on a proposed longer-term fix for reference:
- Currently, the API structure that helps to pull visitation estimates (from `trails-viz-api/trailsvizapi/controller/estimates.py` either references a:
    -  `siteid`(e.g., `/api/sites/<string:siteid>/annualEstimates`) or a 
    - `project` (e.g., `/api/projects/<string:project>/annualEstimates`).
- One proposed approach would be to create a more hierarchical API structure, where each `siteid` is called within a specific `project`, for example: `/api/sites/<string:project>/<string:siteid>/annualEstimates`.
- Then, within the `get_data_from_source()` function in `/trails-viz-api/trailsvizapi/repository/prepare_data.py`, we can grab the data from specific project folders using the project name passed in by the API call.

**Comment by sgwinder** [comment]
Following discussion, I agree that we should patch for now. @jblehr will remove the `OKW_000` and `MBS_000` rows from the four csvs in the `trails-viz-data/projects/Coronado` folder on the VM.

A note about the proposed longer term fix: I am not sure that it will solve this issue, since I'm ~90% confident that the "project" referred to in 

> `project` (e.g., `/api/projects/<string:project>/annualEstimates`)

Is actually referring to the `Prjct_codes` (e.g., `WestCascades_MtnLoop` or `WestCascades_MiddleFork`) which is already nested within the `trails-viz-data/projects/WestCascades` folder. These `Prjct_codes` are defined in the polygon file, and then they're used to identify which `siteids` are relevant.

**Comment by jblehr** [comment]
@sgwinder, unfortunately, it looks like this was only a partial fix. When looking at the weekly project-level estimates for `EastCascades`, the dates still extend too far out, even though there is no actual data there (all values are zero).

![image](https://github.com/OutdoorRD/trails-viz/assets/64438787/3cecbc65-2772-42a3-8d4b-ec2ea9124285)

**Comment by sgwinder** [comment]
Hmm. I looked into some of the individual trails and it only seems to be a problem for a few of them. 

But this one is potentially instructive:
<img width="931" alt="image" src="https://github.com/OutdoorRD/trails-viz/assets/46452196/901565eb-d712-4354-8ed9-f4b173296942">

It looks like the only data that is coming through in 2022 is 'onsite' and 'modeled'. These two datasets would be coming from the `wwwir.csv` file - can you confirm that there are no lingering `OKW_000` rows in that one?

**Comment by jblehr** [comment]
Ok, it appears these data are in the `EastCascades` files themselves, and this may have been an existing bug. For example, when checking Icicle Ridge (OKW_379):
```
jblehr@jblehr-mac EastCascades % grep OKW_379 viz_model_wwwir.csv | grep 2022 
OKW_379,21,5,2022,1476,1,7.29768,2022-05-28
OKW_379,23,6,2022,727,2,6.59031,2022-06-11
OKW_379,26,6,2022,459,2,6.13115,2022-07-02
OKW_379,28,7,2022,643,2,6.46695,2022-07-16
OKW_379,30,7,2022,790,1,6.67388,2022-07-30
OKW_379,31,8,2022,1012,1,6.92105,2022-08-06
OKW_379,35,9,2022,1609,1,7.38426,2022-09-03
```

**Comment by sgwinder** [comment]
Thanks @jblehr . Definitely an existing bug, and probably one that I introduced, sorry!

@EmiliaH and I chatted and think that it would probably be good to remove those rows (being all 2022 data) from the `viz_model_wwwir.csv` file, _if it's easy._ Is this something that would be pretty quick to remove?

Thank you!

**Comment by jblehr** [comment]
Ok, @sgwinder and @EmiliaH, the dashboard is updated again. I will push the corresponding changes to `trails-viz-data`.  @sgwinder, for consistency, I would consider updating the corresponding file or code in `trails-viz` or the relevant project.
